tenseal/cpp/tensors/ckksvector.cpp

```
@@ -570,6 +570,60 @@ CKKSVector& CKKSVector::polyval_inplace(const vector<double>& coefficients) {
     return *this;
 }
 
+CKKSVector CKKSVector::conv2d_im2col(const vector<double>& kernel,
+                                     size_t windows_nb) {
+    CKKSVector new_vec = *this;
+    new_vec.conv2d_im2col_inplace(kernel, windows_nb);
+    return new_vec;
+}
+
+CKKSVector& CKKSVector::conv2d_im2col_inplace(const vector<double>& kernel,
+                                              size_t windows_nb) {
+    vector<double> plain_vec;
+    size_t chunck_size = windows_nb;
+    size_t chuncks_nb = kernel.size();
```

maybe throw an exception if the kernel si empty or windows_nb == 0?